### PR TITLE
fix: edit link broken

### PR DESCRIPTION
**Pull request by lawvs**
- edit link text will cancel link

It is a workaround, it broke format skip by paste

### CODE REVIEW

File: packages/blocks/src/__internal__/rich-text/rich-text.ts [hunk 1]
```diff
@@ -87,35 +87,30 @@ export class RichText extends LitElement {
...
-                retain,
-                delta.ops[1]?.insert.toString() || ''
-              );
+              model.text?.insert(delta.ops[1]?.insert.toString() || '', retain);
             } else {
               // FIXME we must add a noon width space to fix cursor
-              this.quill.insertEmbed(retain, 'text', ' ');
+              model.text?.insert(' ', retain);
+
               this.quill.setSelection(retain + 1, 0, 'api');
```
Comment (lawvs) [inline packages/blocks/src/__internal__/rich-text/rich-text.ts:98]: Is `quill.setSelection` recommended?

Comment (doodlewind) [reply packages/blocks/src/__internal__/rich-text/rich-text.ts:97]: I think it's good to have a better reusable `focusRichText`, its API could be like `focusRichText(id: string, index?: number, async = false)`, which could be implemented by using `quill.setSelection` for now. Updating selection state doesn't mutate model, so using quill API is fine here. And this is also required in #161 .

cc @SaikaSakura @DiamondThree

File: packages/store/src/text-adapter.ts [hunk 2]
```diff
@@ -181,6 +185,15 @@ export class Text {
       this._yText.meta = { delete: true };
     });
   }
+  // eslint-disable-next-line @typescript-eslint/ban-types
+  cover(index: number, length: number, content: string, attributes?: Object) {
```
Comment (lawvs) [inline packages/store/src/text-adapter.ts:189]: Maybe `replace` is better to understand?

**Review by lawvs — COMMENTED** [review]
Can you add some comments to explain what the `text-change` do?

And is needed to off the listener at `disconnectedCallback`?

File: packages/store/src/text-adapter.ts [hunk 3]
```diff
@@ -181,6 +185,15 @@ export class Text {
...
+    this._transact(() => {
+      this._yText.delete(index, length);
+      this._yText.insert(index, content, attributes);
+      // @ts-ignore
```
Comment (doodlewind) [inline packages/store/src/text-adapter.ts:193]: Don't use `@ts-ignore` here, check out what we are doing here https://github.com/toeverything/blocksuite/blob/master/packages/store/src/text-adapter.ts#L92-L107

File: packages/store/src/text-adapter.ts (hunk 2 above)
Comment (doodlewind) [reply packages/store/src/text-adapter.ts:189]: +1 for `replace`

**Review by SaikaSakura — APPROVED** [review]

**Pull request merged**